src/Core/Domain/Configuration/ShopConfigurationInterface.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Configuration;
+
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+
+interface ShopConfigurationInterface extends ConfigurationInterface
+{
+    /**
+     * @param string $key
+     * @param mixed|null $default
+     * @param ShopConstraint|null $shopConstraint
+     *
+     * @return mixed
+     */
+    public function get($key, $default = null, ShopConstraint $shopConstraint = null);
+
+    /**
+     * @param string $key
+     * @param mixed $value
+     * @param ShopConstraint|null $shopConstraint
+     *
+     * @return mixed
+     */
+    public function set($key, $value, ShopConstraint $shopConstraint = null);
```

Comment:
```suggestion
     * @return ShopConfigurationInterface
     */
    public function set($key, $value, ShopConstraint $shopConstraint = null): ShopConfigurationInterface;
```

Reply:
I can't add any strict typing because it wouldn't respect the parent interface `ConfigurationInterface`
Already tried.. got kicked!

Reply:
I can only improve the PHPDoc